File: Array/3190. Find Minimum Operations to Make All Elements Divisible by Three/minimum_operations.cpp
#include <vector>
#include <iostream>

class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        int result=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]%3!=0){
                result++;
            }
        }
        return result;
    }
};

int main() {
    Solution sol;
    std::vector<int> nums1 = {1, 2, 3, 4};
    std::vector<int> nums2 = {3, 6, 9};

    std::cout << "Minimum operations for nums1: " << sol.minimumOperations(nums1) << std::endl;
    std::cout << "Minimum operations for nums2: " << sol.minimumOperations(nums2) << std::endl;

    return 0;
}

File: Array/3190. Find Minimum Operations to Make All Elements Divisible by Three/minimum_operations.rs
struct Solution;

impl Solution {
    pub fn minimum_operations(nums: Vec<i32>) -> i32 {
        let mut result=0;
        for num in nums.iter(){
            if num%3!=0 {
                if (num+1)%3==0 || (num-1)%3==0 {
                    result+=1
                }
            }
        }
        result
    }
    pub fn minimum_operations1(nums: Vec<i32>) -> i32 {
        let mut result = 0;
        for &num in nums.iter() {
            match num % 3 {
                1 | 2 => result += 1, // 如果num对3取模的结果是1或2，增加操作计数
                _ => (), // 如果num能被3整除，不做任何操作
            }
        }
        result
    }
}

fn main(){
    let nums1 = vec![1, 2, 3, 4];
    let nums2 = vec![3, 6, 9];

    println!("Minimum operations for nums1: {}", Solution::minimum_operations(nums1));
    println!("Minimum operations for nums2: {}", Solution::minimum_operations(nums2));
}